swap/src/cli/watcher.rs

```diff
@@ -0,0 +1,100 @@
+use std::sync::Arc;
+use std::collections::HashMap;
+use std::time::Duration;
+
+use anyhow::Result;
+use uuid::Uuid;
+
+use crate::bitcoin::{ExpiredTimelocks, Wallet};
+use crate::cli::api::tauri_bindings::TauriHandle;
+use crate::protocol::bob::BobState;
+use crate::protocol::{Database, State};
+
+use super::api::tauri_bindings::TauriEmitter;
+
+/// A long running task which watches for changes to timelocks and the number of confirmations.
+#[derive(Clone)]
+pub struct Watcher {
+    wallet: Arc<Wallet>,
+    database: Arc<dyn Database + Send + Sync>,
+    /// This saves for every running swap the expired timelocks as well as
+    /// the [`ScriptStatus`] of [`TxLock`].
+    current_swaps: HashMap<Uuid, ExpiredTimelocks>,
+    tauri: Option<TauriHandle>,
+}
+
+impl Watcher {
+    /// How often to check for changes (in seconds)
+    const CHECK_INTERVAL: u64 = 3;
```

Comment:
Increasing this to 30-60 seconds would be enough imo. The status of the script can change at most every minute.